### PR TITLE
Generate GenerateAutomaticInterfaceAttribute

### PR DESCRIPTION
**Pull request by crwsolutions**
Generate GenerateAutomaticInterfaceAttribute instead of asking that of the user.

See cookbook:
https://github.com/dotnet/roslyn/blob/main/docs/features/incremental-generators.cookbook.md#generated-class

If you're feeling adventurous, it is possible to remove the namespace from the generated attribute (so remove 'namespace AutomaticInterface'). That would give even more convenience. Given the name of the attribute I think there is very little change of type conflicts.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -126,9 +126,8 @@ namespace AutomaticInterfaceExample
 ## How to use it?
```
[the remark this answers is not in the document]
Comment (crwsolutions) [reply README.md:126]: ✅

File: README.md [hunk 2]
```diff
@@ -126,9 +126,8 @@ namespace AutomaticInterfaceExample
...
 
 1. Install the nuget: `dotnet add package AutomaticInterface`
-2. Create an Attribute with the Name `[GenerateAutomaticInterface]`. You can just copy the minimal code from this Repo (see the `AutomaticInterfaceAttribute` project). It's the easiest way to get that attribute because you cannot reference any code from the analyzer package.
```
[the remark this answers is not in the document]
Comment (crwsolutions) [reply README.md:129]: I think because it is an analyzer it is immediate available, so a build is not necessary. But I have rewritten it.

**Comment by crwsolutions** [comment]
Just to dust off this PR :): Hi, I've added a step in the README, to add the using statement. So if you are happy with it this can be merged now in my opinion. But it is also possible to remove the namespace for some extra convenience (would be mine preference). I would also like to highlight in that case, that the generated Attribute is internal. So the probability of conflicts is really about zero.

**Comment by ccHanibal** [comment]
Any chance that this gets merged and released in a foreseeable time frame @ChristianSauer? I would really like to start working on some new features (like custom interface name and get only properties in interfaces), but I consider this a prerequisite.

**Comment by ChristianSauer** [comment]
@ccHanibal @crwsolutions I've finally merged it. Sorry, where a shitty few months

**Pull request merged**